generator/internal/language/rust.go

```diff
@@ -506,12 +506,45 @@ func rustFieldType(f *api.Field, state *api.APIState, primitive bool, modulePath
 	if !primitive && f.Repeated {
 		return fmt.Sprintf("std::vec::Vec<%s>", rustBaseFieldType(f, state, modulePath, sourceSpecificationPackageName, packageMapping))
 	}
+	if !primitive && f.Recursive {
```

Comment:
nit: all of the top level if checks check `!primitive`. Maybe this should be factored out.

Reply:
Done.